src/v/iceberg/values_avro.cc

```diff
@@ -113,14 +128,28 @@ struct primitive_value_avro_visitor {
         }
         return {data};
     }
-    avro::GenericDatum operator()(const decimal_value&) {
-        throw std::invalid_argument("XXX decimal not implemented");
+    avro::GenericDatum operator()(const decimal_value& v) {
+        std::vector<uint8_t> bytes(16);
+        auto low = absl::Int128Low64(v.val);
+        auto high = absl::Int128High64(v.val);
+
+        // Store in little-endian order (least significant byte first)
+        for (size_t i = 0; i < 8; ++i) {
+            bytes[i] = static_cast<uint8_t>(low >> (i * 8));
+            bytes[i + 8] = static_cast<uint8_t>(high >> (i * 8));
+        }
+
+        return {avro_schema_, avro::GenericFixed(avro_schema_, bytes)};
     }
-    avro::GenericDatum operator()(const fixed_value&) {
-        throw std::invalid_argument("XXX fixed not implemented");
+    avro::GenericDatum operator()(const fixed_value& fixed) {
+        std::vector<uint8_t> bytes(fixed.val.size_bytes());
+        iobuf::iterator_consumer it(fixed.val.cbegin(), fixed.val.cend());
+        it.consume_to(fixed.val.size_bytes(), bytes.data());
+        return {avro_schema_, avro::GenericFixed(avro_schema_, bytes)};
     }
-    avro::GenericDatum operator()(const uuid_value&) {
-        throw std::invalid_argument("XXX uuid not implemented");
+    avro::GenericDatum operator()(const uuid_value& uuid) {
+        maybe_throw_invalid_schema(avro_schema_, avro::AVRO_STRING);
+        return {avro_schema_, fmt::to_string(uuid.val)};
```

Comment:
The avro spec mentions this should be a 16-byte string/fixed

>A uuid logical type annotates an Avro string or fixed of length 16. Both the string and fixed byte layout have to conform with [RFC-4122](https://www.ietf.org/rfc/rfc4122.txt).

Reply:
I read this as string OR fixed (length 16). Mihal just choose string here, which is fine IMO. We probably need to handle reading both

Reply:
Ahh I think you're right, they do just call fromString() here:

https://github.com/apache/avro/blob/80400781a796bc0e90dd8ea1db42234926db33e9/lang/java/avro/src/main/java/org/apache/avro/Conversions.java#L63-L65

Reply:
The spec seems a bit inconsistent with the CPP code: 
https://github.com/redpanda-data/avro/blob/1410e79f9df61669c2d52f6d0643e6c35156e615/lang/c%2B%2B/impl/Node.cc#L139

Reply:
Yeah, the C++ library is definitely lagging in completeness relative to other languages